This patch is only needed and only works with wxWidgets >= 2.9.5.
This should be fixed upstream, but a better patch is needed for backward compatibility.

--- src/plugins/contrib/source_exporter/wxPdfDocument/include/wx/pdfdc29.h.orig
+++ src/plugins/contrib/source_exporter/wxPdfDocument/include/wx/pdfdc29.h
@@ -162,9 +162,9 @@
   virtual void DoGetSize(int* width, int* height) const;
   virtual void DoGetSizeMM(int* width, int* height) const;
 
-  virtual void DoDrawLines(int n, wxPoint points[],
+  virtual void DoDrawLines(int n, const wxPoint points[],
                            wxCoord xoffset, wxCoord yoffset);
-  virtual void DoDrawPolygon(int n, wxPoint points[],
+  virtual void DoDrawPolygon(int n, const wxPoint points[],
                              wxCoord xoffset, wxCoord yoffset,
                              wxPolygonFillMode fillStyle = wxODDEVEN_RULE);
   virtual void DoDrawPolyPolygon(int n, int count[], wxPoint points[],
--- src/plugins/contrib/source_exporter/wxPdfDocument/include/wx/pdfprint.h.orig
+++ src/plugins/contrib/source_exporter/wxPdfDocument/include/wx/pdfprint.h
@@ -867,7 +867,7 @@
     return rval;
   }
 
-  virtual void DoDrawLines(int n, wxPoint points[],
+  virtual void DoDrawLines(int n, const wxPoint points[],
                            wxCoord xoffset, wxCoord yoffset)
   {
     m_dc.DoDrawLines(n, points,xoffset, yoffset);
@@ -874,7 +874,7 @@
     UpdateBoundingBox();
   }
 
-  virtual void DoDrawPolygon(int n, wxPoint points[],
+  virtual void DoDrawPolygon(int n, const wxPoint points[],
                              wxCoord xoffset, wxCoord yoffset,
                              wxPolygonFillMode fillStyle = wxODDEVEN_RULE)
   {
--- src/plugins/contrib/source_exporter/wxPdfDocument/src/pdfdc29.inc.orig
+++ src/plugins/contrib/source_exporter/wxPdfDocument/src/pdfdc29.inc
@@ -1032,7 +1032,7 @@
 }
 
 void
-wxPdfDCImpl::DoDrawLines(int n, wxPoint points[], wxCoord xoffset, wxCoord yoffset)
+wxPdfDCImpl::DoDrawLines(int n, const wxPoint points[], wxCoord xoffset, wxCoord yoffset)
 {
   wxCHECK_RET(m_pdfDocument, wxT("Invalid PDF DC"));
   SetupPen();
@@ -1039,7 +1039,7 @@
   int i;
   for (i = 0; i < n; ++i)
   {
-    wxPoint& point = points[i];
+    const wxPoint& point = points[i];
     double xx = ScaleLogicalToPdfX(xoffset + point.x);
     double yy = ScaleLogicalToPdfY(yoffset + point.y);
     CalcBoundingBox(point.x+xoffset, point.y+yoffset);
@@ -1056,7 +1056,7 @@
 }
 
 void
-wxPdfDCImpl::DoDrawPolygon(int n, wxPoint points[],
+wxPdfDCImpl::DoDrawPolygon(int n, const wxPoint points[],
                            wxCoord xoffset, wxCoord yoffset,
                            wxPolygonFillMode fillStyle /* = wxODDEVEN_RULE*/)
 {
@@ -1068,7 +1068,7 @@
   int i;
   for (i = 0; i < n; ++i)
   {
-    wxPoint& point = points[i];
+    const wxPoint& point = points[i];
     xp.Add(ScaleLogicalToPdfX(xoffset + point.x));
     yp.Add(ScaleLogicalToPdfY(yoffset + point.y));
     CalcBoundingBox(point.x + xoffset, point.y + yoffset);
